src/main/java/net/dv8tion/jda/internal/entities/TemplateImpl.java

```diff
@@ -0,0 +1,258 @@
+    {
+        return this.description;
+    }
+
+    @Override
+    public int getUses()
+    {
+        return this.uses;
+    }
+
+    @Nonnull
+    @Override
+    public User getCreator()
+    {
+        return this.creator;
+    }
+
+    @Nonnull
+    @Override
+    public OffsetDateTime getTimeCreated()
+    {
+        return this.createdAt;
+    }
+
+    @Nonnull
+    @Override
+    public OffsetDateTime getTimeUpdated()
+    {
+        return this.updatedAt;
+    }
+
+    @Nonnull
+    @Override
+    public Template.Guild getGuild()
+    {
+        return this.guild;
+    }
+
+    @Override
+    public boolean isSynced()
+    {
+        return this.synced;
+    }
+
+    @Nonnull
+    @Override
+    public TemplateManager getManager()
+    {
+        TemplateManager mng = manager;
+        if (mng == null)
+        {
+            mng = MiscUtil.locked(mngLock, () ->
+            {
+                if (manager == null)
+                    manager = new TemplateManagerImpl(this);
+                return manager;
+            });
+        }
+        return mng;
+    }
+
+    private void checkInteraction()
+    {
+        final net.dv8tion.jda.api.entities.Guild guild = this.api.getGuildById(this.guild.getIdLong());
+
+        if (guild == null)
+            throw new IllegalStateException("Cannot interact with a template without shared guild");
+        if (!guild.getSelfMember().hasPermission(Permission.MANAGE_SERVER))
+            throw new InsufficientPermissionException(guild, Permission.MANAGE_SERVER);
+    }
+
+    @Nonnull
+    @Override
+    public JDA getJDA()
+    {
+        return this.api;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return code.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (obj == this)
+            return true;
+        if (!(obj instanceof TemplateImpl))
+            return false;
+        TemplateImpl impl = (TemplateImpl) obj;
+        return impl.code.equals(this.code);
+    }
+
+    @Override
+    public String toString()
+    {
+        return "Template(" + this.code + ")";
+    }
+
+    public static class GuildImpl implements Guild
+    {
+        private final String name, iconId;
+        private final long id;
+        private final VerificationLevel verificationLevel;
+
+        public GuildImpl(final long id, final String iconId, final String name, final VerificationLevel verificationLevel)
+        {
+            this.id = id;
+            this.iconId = iconId;
+            this.name = name;
+            this.verificationLevel = verificationLevel;
+        }
+
+        @Nullable
+        @Override
+        public String getIconId()
+        {
+            return this.iconId;
+        }
+
+        @Nullable
+        @Override
+        public String getIconUrl()
+        {
+            return this.iconId == null ? null
+                    : "https://cdn.discordapp.com/icons/" + this.id + "/" + this.iconId + ".png";
```

Comment:
Make this a constant in the interface and use `String.format`

Reply:
there is a constant for this in Guild interface already, do i just use that one or you want me to create another constant in Template interface?

Reply:
If its the same endpoint, use it.